client/src/pages/forms/SignupForm.js

```diff
@@ -0,0 +1,162 @@
+	}
+	useEffect(() => {
+		console.log("errors", errors);
+		if (Object.keys(errors).length === 0 && submit) {
+			setDetails(initialDetails);
+			console.log("details entered:", details);
+		}
+	}, [details, errors, initialDetails, submit]);
+
+	const validate = (details) => {
+		const errors = {};
+		const userNameRegex = /^[a-z0-9]+(?:[ _.-][a-z0-9]+)*$/;
+
+
+		if (!details.Name) {
+			errors.Name = "Name is required";
+		} else if (details.Name.length < 3) {
+			errors.Name = "Please enter your full name";
+		}
+		if (!details.Class) {
+			errors.Class = "Class is required";
+		} else if (details.Class.length < 3 || details.Class.length > 7) {
+			errors.Class = "Please enter the correct Class. i.e: WM3, NW4";
+		}
+		if (!details.Username) {
+			errors.Username = "Username is required";
+		} else if (
+			details.Username.length < 3 ||
+			!userNameRegex.test(details.Username)
+		) {
+			errors.Username = "Required format: word, word-sep-word";
+		}
+		if (!details.Password) {
+			errors.Password = "Password is required";
+		} else if (details.Password.length < 3) {
+			errors.Password = "Password is too short";
+		}
+		if (details.ConfirmPassword !== details.Password) {
+			errors.ConfirmPassword = "Passwords do not match";
+		}
+
+		return errors;
+	};
+	return (
+		<section className="signup__forms">
+			<div className="trainee__form">
+
+				{Object.keys(errors).length === 0 && submit ? (
+					<div className="ui msg success">Signed Up Successfully</div>
+				) : (
+					""
+				)}
+				<form onSubmit={submitHandler}>
+					<div className="form-inner">
+						<h2>Sign Up Page</h2>
+						<h3>Please enter your details as a trainee</h3>
+
+						<div className="form-group">
+							<label htmlFor="trainee_name">Name:</label>
+							<input
+								type="text"
+								name="trainee_name"
+								id="trainee_name"
+								onChange={(e) =>
+									setDetails({ ...details, Name: e.target.value })
+								}
+								value={details.Name}
+							/>
+						</div>
+						<p className="form__error">{errors.Name}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_name">Class:</label>
+							<input
+								type="text"
+								name="trainee_class"
+								id="trainee_class"
+								onChange={(e) =>
+									setDetails({ ...details, Class: e.target.value })
+								}
+								value={details.Class}
+							/>
+						</div>
+						<p className="form__error">{errors.Class}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_Username">Username:</label>
+							<input
+								type="text"
+								name="trainee_Username"
+								id="trainee_Username"
+								onChange={(e) =>
+									setDetails({ ...details, Username: e.target.value })
+								}
+								value={details.Username}
+							/>
+						</div>
+						<p className="form__error">{errors.Username}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_name">Password:</label>
+							<input
+								type="trainee_Password"
```

Comment:
I'm not sure about this type here. There is a input type of "password", I'm not sure about this type. Same as below for the confirm password input